aya-obj/src/obj.rs

```diff
@@ -898,47 +983,25 @@ impl Object {
             BpfSectionKind::Text => self.parse_text_section(section)?,
             BpfSectionKind::Btf => self.parse_btf(&section)?,
             BpfSectionKind::BtfExt => self.parse_btf_ext(&section)?,
-            BpfSectionKind::BtfMaps => {
-                let symbols: HashMap<String, Symbol> = self
-                    .symbol_table
-                    .values()
-                    .filter(|s| {
-                        if let Some(idx) = s.section_index {
-                            idx == section.index.0 && s.name.is_some()
-                        } else {
-                            false
-                        }
-                    })
-                    .cloned()
-                    .map(|s| (s.name.as_ref().unwrap().to_string(), s))
-                    .collect();
-                self.parse_btf_maps(&section, symbols)?
-            }
+            BpfSectionKind::BtfMaps => self.parse_btf_maps(&section)?,
             BpfSectionKind::Maps => {
                 // take out self.maps so we can borrow the iterator below
                 // without cloning or collecting
                 let mut maps = mem::take(&mut self.maps);
 
                 // extract the symbols for the .maps section, we'll need them
                 // during parsing
-                let symbols = self.symbol_table.values().filter(|s| {
-                    s.section_index
-                        .map(|idx| idx == section.index.0)
-                        .unwrap_or(false)
-                });
+                let symbols = self.symbols_by_section.get(&section.index).unwrap().iter();
```

Comment:
We shouldn't unwrap here

Say I'm bpfd and load a bunch of user provided .o files

Say I'm an attacker and I manage to put a specially crafted .o file in the bpfd
load path

I can strip .maps symbols from the symbol table to crash the whole bpfd process. 

In general, it's ok to unwrap() on things that can't fail by construction, for instance
the symbol.name.unwrap() you do where symbols with None names are pre-filtered
(although in that case expect() is better than unwrap()). It's not ok to unwrap
on invariants that depend on external inputs.